### DISCUSSION TITLE
Auswahl Auto

### DISCUSSION THREAD

**Discussion by Trunks1982** [camueller/SmartApplianceEnabler#241] [Installation und Betrieb]
Hi ich habe mir gerade die SOC Abfrage für ein Auto eingerichtet, was auch soweit funktioniert.
Jetzt habe ich zwei Auto unter einer Wallbox eingerichtet, das eine Auto macht die SOC Abfrage das andere soll die Standardwerte von 0%-100% übernehmen.Wenn ich jetzt das Auto auswähle was nicht diese SOC Abfrage macht, bleibt der Wert von dem Auto was die SOC Abfrage macht.Soll das so sein oder ist das nen Bug?

**Comment by camueller** [comment]
Die Werte werden erst zurückgesetzt, wenn Du das Auto von der Wallbox trennst und neu verbindest. Hast Du das gemacht?

**Reply by Trunks1982** [reply]
Nein.Aber wenn ich jetzt zb dann das andere auto anschliesse wo keine soc abfrage gemacht wird müsste ich ja erst das andere Auto im sae aussuchen ne? Könnte man das nicht so machen wenn schon angeschlossen dann das auto aussuchen?Wäre das nicht sinniger bzw auch einfacher?

**Reply by camueller** [reply]
Doch, das Aussuchen machst Du nach dem Verbinden, wenn Du Dich für einen Lademodus entscheidest. Trotzdem erfolgt das Zurücksetzen der Werte beim Trennen/Verbinden mit der Wallbox.

**Reply by Trunks1982** [reply]
Mhm also erst verbinden aussuchen und dann nochmal trennen und verbinden?

**Reply by camueller** [reply]
1. Nicht verbunden
2. Fahrzeug verbinden
3. Fahrzeug/Lademodus auswählen
4. Fahrzeug wird geladen entsprechend Konfiguration

**Reply by Trunks1982** [reply]
Ok.Teste es gerade und es wird immer das erste Fahrzeug genommen wo die SOC Abfrage gemacht wird ausgesucht.Änder ich dann zb den Lademodis auch auf pv überschuss und wähle dann das andere  Auto bleibt der SoC vom ersten Fahrzeug drin.Obwohl standart 0-100% beim zweiten Fahrzeug hinterlegt ist.Sollte ja dann nach deiner Anleitung wechseln.Passiert aber nicht.

**Reply by camueller** [reply]
Dann ist da wohl ein Fehler drin.

**Reply by Trunks1982** [reply]
Genau, da wollte ich nur sicher gehn.Kannst ja mal selber testen.Müsste ja dann bei dir auch so vorkommen.

**Reply by camueller** [reply]
Habe mal einen Bug erstellt dafür: https://github.com/camueller/SmartApplianceEnabler/issues/242

**Comment by Trunks1982** [comment]
Was mir noch aufgefallen ist, wenn ich zb ne Zeit für  das updaten vom SOC einstelle funktioniert dies auch nicht.Im Shell zeigt er mir dann den richtigen Wert an.Wird aber nicht im SAE übernommen.

**Reply by solanjo** [reply]
Hallo, das kann ich bestätigen. Bei mir kann eine einmal eingegebene Zeit und/oder SoC fürs Update nicht mehr verändert werden. Der Button "Speichern" wird zwar aktiv wenn man diese Werte verändert, aber die Änderung wird dann nicht abgespeichert.
An der Stelle hab ich mir beholfen, die Werte direkt in die Appliances.xml einzutragen. Unabhängig davon scheint mir der SAE aber anhand der eingegebenen Daten trotzdem kein Update des tatsächlichen SoC zu machen. Beim Anstecken des Fahrzeugs wird der richtige Wert ausgelesen. Finde dann aber im Log keine weiteren Einträge (habe auf 00:30 und 2% Änderung gestellt, was sicher eher übertrieben ist).

**Reply by camueller** [reply]
Vermutlich meinst Du den Lademodus "Optimiert", weil man nur bei disem Tag/Zeit eingeben kann?

**Reply by solanjo** [reply]
Nein, ich meinte die Ausführungswiederholung des SoC-Scripts.

Allerdings muss ich nun zurückrudern - hatte beschriebenes Phänomen unter der 1.6.19 vor einiger Zeit festgestellt. Mit der 1.6.20 eben getestet geht es jetzt!

**Reply by Trunks1982** [reply]
Ich nutze die 1.6.20 und bei mir funktionierts nichts.

**Reply by solanjo** [reply]
Hast Du mal im Log geschaut, ob der SAE über die Regex den richtigen Wert ausliest?

**Reply by Trunks1982** [reply]
Anfangs wenn ich das Auto anschliesse funktionierts ja.Ka warum hinterher nicht mehr.

**Comment by Trunks1982** [comment]
wie müsste der Befehl im log aussehen wenn er den Soc abfragt?
Sitze die ganze Zeit vom Log und kann nirgends den Befehl für das Script sehn.

**Reply by camueller** [reply]
So:
```
2022-04-03 14:18:43,529 DEBUG [Thread-422] d.a.s.c.e.SocScript [SocScript.java:108] F-00000001-000000000019-00: Executing SoC script: /opt/sae/soc/soc.sh
...
2022-04-03 14:19:33,624 DEBUG [Thread-422] d.a.s.c.e.SocScript [SocScript.java:118] F-00000001-000000000019-00: SoC script output: Prepare Session
Login...
state_of_charge 78
```

**Reply by Trunks1982** [reply]
Ok.Hab nachgeguckt und es wird nur einmal nach anschliessen des Autos aufgerufen.Ich kann dir gerne mal mein log schicken wenn du willst.Mein Abfrage Intervall hab ich auf 10min eingestellt.

**Reply by camueller** [reply]
Schau bitte in die Doku, da steht, wann und wie oft das Script per Default aufgerufen wird.

**Reply by Trunks1982** [reply]
Ah jetzt klappt's.Also müssen beide Werte eingetragen werden für die Abfrage.
